lib/sync-server.js

```diff
@@ -100,7 +100,12 @@ var DEFAULT_SYNC_CONF = {
   /**@type {String} specify the maximum retention time of an inactive datasetClient. Any inactive datasetClient that is older than this period of time will be removed.*/
   datasetClientCleanerRetentionPeriod: '24h',
   /** @type {String} specify the frequency the datasetClient cleaner should run. Default every hour.*/
-  datasetClientCleanerCheckFrequency: '1h'
+  datasetClientCleanerCheckFrequency: '1h',
+
+  /** @type {Boolean} Specify if the server should wait for the ack insert to complete before returning the response for the sync request. Default is true. */
+  syncReqWaitForAck: true,
+  /** @type {Number} Specify the max number of ack items will be processed for a single request. Default is -1 (unlimited).*/
+  syncReqAckLimit: -1
```

Comment:
@wtrocki yeah, good question. Let me try to explain it.

So an acknowledgement is generated when a client submits a pending change and that pending change is getting processed by the sync framework. The result of that is saved to the `fhsync_<dataset_name>_updates` collection, and that is effectively the `acknowledgement`.  When the same client make a sync request the next time, the sync server will return those updates back to the client and the client can use those to remove the pending changes from its own dataset. Then the client will send the acks back to the server again, so that server can remove the records from the `fhsync_<dataset_name>_updates` collection.

So when we only process a subset of the acks in the requests, it is ok as the rest of the `acks` will be still in the `fhsync_<dataset_name>_updates` collection, and the client will get them again and send the rest.

For example, if there are 100 updates (acks) for a client in the db, and the client is sending all 100 ack in the first sync request.

In the first request, we only process the first 10. They will be put them on the ack queue and discard the rest. In this same request, the client will get the next 10 updates back (see https://github.com/feedhenry/fh-sync/blob/master/lib/api-sync.js#L76-L114).

In the next sync request, the client will only send the 10 updates. But this time because the first 10 are processed in the queue, the client will get the next batch of 10 updates back.

So this process will run over and over again. But because all the clients' acks are returned from the server, as long as we not deleting anything from the `updates` collection, there will no data lost.

But let's say in the worst case, there are `acks` lost, that is ok as well. Because the purpose of the `acks` is to remove the `updates` from db. Lost some `acks` just mean we have redundant data in the db and we might have to clean them  manually.